core/components/organisms/navigation/HorizontalNavigation.tsx

```diff
@@ -0,0 +1,97 @@
+import * as React from 'react';
+import classNames from 'classnames';
+import { Text, Icon, Pills } from '@/index';
+import { NavigationProps } from '@/index.type';
+import {
+  getTextAppearance,
+  getIconAppearance,
+  getPillsAppearance,
+  isActive,
+  Menu,
+} from './utils';
+
+export interface HorizontalNavigationProps {
+  menus: NavigationProps['menus'];
+  active?: NavigationProps['active'];
+  onClick?: NavigationProps['onClick'];
+}
+
+export const HorizontalNavigation = (props: HorizontalNavigationProps) => {
+  const {
+    menus,
+    active,
+    onClick,
+  } = props;
+
+  const onClickHandler = (menu: Menu) => {
+    if (onClick) onClick(menu);
+  };
+
+  const getPillsClass = (disabled?: boolean) => {
+    return classNames({
+      ['Navigation-horizontalPills']: true,
+      ['Navigation-horizontalPills--disabled']: disabled
+    });
+  };
+
+  const renderIcon = (menu: Menu, isMenuActive: boolean) => {
+    if (menu.count !== undefined) {
+      const count = menu.count > 99 ? '99+' : menu.count;
+      return (
+        <Pills
+          subtle={menu.disabled}
+          className={getPillsClass(menu.disabled)}
+          appearance={getPillsAppearance(isMenuActive)}
+          data-test="DesignSystem-HorizontalNavigation--Pills"
+        >
+          {count}
+        </Pills>
+      );
+    }
+
+    if (menu.icon) {
+      return (
+        <Icon
+          className="mr-3"
+          name={menu.icon}
+          appearance={getIconAppearance(isMenuActive, menu.disabled)}
+          data-test="DesignSystem-HorizontalNavigation--Icon"
+        />
+      );
+    }
+
+    return null;
+  };
+
+  const list = menus.map((menu, index) => {
+    const isMenuActive = isActive(menu, menus, active);
```

Comment:
I think menus should be the first argument here. 
@satyamyadav What's your opinion on the same?

Reply:
Yes.